### PR TITLE
display user's current location on map

### LINKED ISSUE #22: Get current latitude and longitude from geolocation (Google Map)

### PR DESCRIPTION
**Pull request by SachiGoto**
### Feature Description

Display a user's current location on Google Maps. Essentially, a prompt will pop up, and users need to allow Google Maps to detect their current location.

Screenshot 2023-07-17 at 10 56 27 AM

[Documentation I referred ](https://developers.google.com/maps/documentation/javascript/geolocation)

I set the default location to be Downtown Vancouver, so if users choose 'Block,' the default location will be displayed on the map instead.

### Something you want to mention
I noticed that sometimes the map shows in light mode while the header is in dark mode. I'm not really sure why. I console logged to see what the current mode is, and sometimes when it's dark mode in the header, it says light mode in console.log. Is this some kind of rendering issue?

### Checklist before requesting a review

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings

### CODE REVIEW

File: src/app/map/page.tsx
Comment (Lada496) [inline src/app/map/page.tsx:1]: Can we remove these comments?

Comment (SachiGoto) [reply src/app/map/page.tsx:1]: Sorry what are these comments??

File: src/app/map/page.tsx [hunk 1]
```diff
@@ -8,9 +8,9 @@ export const metadata: Metadata = {
 
 const MapPage = () => {
   // [TODO] get current latitude and longitude from geolocation
```
Comment (Lada496) [inline src/app/map/page.tsx]:
```suggestion
```

File: src/app/map/page.tsx [hunk 2]
```diff
@@ -8,9 +8,9 @@ export const metadata: Metadata = {
...
-  const address = "601 W Cordova St, Vancouver, BC V6B 1G1";
+  // const address = "601 W Cordova St, Vancouver, BC V6B 1G1";
```
Comment (Lada496) [inline src/app/map/page.tsx]:
```suggestion
```

File: src/app/map/page.tsx
Comment (Lada496) [reply src/app/map/page.tsx:1]: I'm sorry for mentioning them. Please find my suggestion below. 
There are 2 comments I think we can remove🙌

Comment (SachiGoto) [reply src/app/map/page.tsx:1]: Ah, I see! Thank you for pointing that out!  I just deleted those comments!

Comment (Lada496) [reply src/app/map/page.tsx:1]: Thank you!

**Review by Lada496 — APPROVED** [review]
Thank you for setting up to get current users location!
I think we can discuss how we can work on map loading state with #39

**Review by devjoylee — APPROVED** [review]
LGTM. I added the related issue #22 to be closed together with this PR.

**Pull request merged**